### ISSUE TITLE
Document generator bug with extended classes

### ISSUE THREAD

**Issue by michaelperrin** [doctrine/mongodb-odm#1299]
Suppose you have these 2 classes:

```php
abstract class AbstractUser
{
    /** @ODM\Id */
    protected $id;
    
    /** @ODM\Field(type="string") */
    protected $name;
    
    public function getId()
    {
        return $this->id;
    }
    
    public function getName()
    {
        return $this->name;
    }
    
    public function setName($name)
    {
        $this->name = $name;
    }
}
```

and:

```php
class User extends AbstractUser
{
    /** @ODM\Field(type="string") */
    protected $lastname;
}
```

Then when generating documents, the fields, getters and setters from `AbstractUser` are duplicated to the `User` child class, which should not be the case.

A PR will be opened in the next couple of minutes.

**Comment by michaelperrin** [comment]
Fixed in #1300 .

**Comment by malarzm** [comment]
#1300 was merged manually in https://github.com/doctrine/mongodb-odm/commit/490ca30235f1bc2b1f85b15389a5ba43a47e7a05 and will be merged into `master` branch soon
